apps/meteor/server/models/raw/LivechatContacts.ts

```diff
@@ -195,4 +199,36 @@ export class LivechatContactsRaw extends BaseRaw<ILivechatContact> implements IL
 			options,
 		).toArray();
 	}
+
+	isContactActiveOnPeriod(visitorId: string, period: string): Promise<boolean> {
+		const query = {
+			'channels.visitorId': visitorId,
+			'activity': period,
+		};
+
+		return this.findOne(query, { projection: { _id: 1 } }).then(Boolean);
```

Comment:
what about just counting? that way it would return 0/>0 and the function can work without having to chain the promise

Reply:
Good idea! I just repeated what we were doing before, but that's definitely better